wwwapp/views.py

```diff
@@ -199,6 +199,7 @@ def workshop_view(request, name=None):
         workshop = None
         title = 'Nowe warsztaty'
         if not request.user.is_authenticated:
+            request.session['next'] = request.path
```

Comment:
This is what I call "PKP Intercity bug" - it breaks if you have two tabs open. You should be using a GET parameter instead.

Specifically, it breaks in this scenario:
1. Open website in tab A and tab B
2. Click one link in tab A, get redirected to login page
3. Click another link in tab B, get redirected to login page
4. Login in tab A - the link from tab B opens ?!

Reply:
Let's ignore this problem maybe?

Reply:
Just pass it using a GET parameter like any sane website does... I think it shouldn't be that hard?